docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,585 @@
+Error Details:
+action declares irrelevant authority '{"actor":"bob","permission":"customp2"}'; minimum authority is {"actor":"bob","permission":"active"}
+```
+
+## Delete Custom Permissions
+Now you have unlinked the _customp2_ permission you can delete this permission.
+
+```shell
+cleos set account permission bob customp2 NULL active -p bob@active
+```
+
+The output should be similar to:
+
+```console
+executed transaction: 3f3e58707e5548ec34f5655327b1110c18d455c9ee0a6cffc102d7bc4e0a6cdb  112 bytes  472 us
+#         eosio <= eosio::deleteauth            {"account":"bob","permission":"customp2"}
+```
+Let’s check the account:
+
+```shell
+cleos get account bob --json
+```
+
+The output will be similar to:
+
+```json
+{
+  "account_name": "bob",
+  "head_block_num": 23485,
+  "head_block_time": "2021-05-06T07:46:19.000",
+  "privileged": false,
+  "last_code_update": "1970-01-01T00:00:00.000",
+  "created": "2021-05-06T05:07:27.500",
+  "ram_quota": -1,
+  "net_weight": -1,
+  "cpu_weight": -1,
+  "net_limit": {
+    "used": -1,
+    "available": -1,
+    "max": -1
+  },
+  "cpu_limit": {
+    "used": -1,
+    "available": -1,
+    "max": -1
+  },
+  "ram_usage": 4414,
+  "permissions": [{
+      "perm_name": "active",
+      "parent": "owner",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    },{
+      "perm_name": "customp1",
+      "parent": "active",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    },{
+      "perm_name": "customp3",
+      "parent": "active",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    },{
+      "perm_name": "owner",
+      "parent": "",
+      "required_auth": {
+        "threshold": 1,
+        "keys": [{
+            "key": "EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D",
+            "weight": 1
+          }
+        ],
+        "accounts": [],
+        "waits": []
+      }
+    }
+  ],
+  "total_resources": null,
+  "self_delegated_bandwidth": null,
+  "refund_request": null,
+  "voter_info": null,
+  "rex_info": null
+}
+```
+
+## Conclusion
+
+This tutorial demonstrates how to create, link, and delete custom permissions. The tutorial shows you how to use these permissions to control which actions an account permission can perform.  
+
+## Further Examples
+
+Custom permissions can also be controlled directly using the JavaScript SDK, for examples follow these links:
```

Comment:
Construction:
SDK, for examples -> SDK. For examples

Reply:
updated